### PR TITLE
add validation on model save

### PR DESCRIPTION
**Pull request by benauriemma**

### CODE REVIEW

File: rackcity/models/asset.py [hunk 1]
```diff
@@ -217,6 +217,7 @@ class Meta:
 
     def save(self, *args, **kwargs):
         try:
+            self.full_clean()
```
Comment (juliasaveliff) [inline rackcity/models/asset.py:220]: what is this?

File: rackcity/utils/change_planner_utils.py [hunk 2]
```diff
@@ -76,21 +79,41 @@ def detect_conflicts_cp(sender, **kwargs):
...
-                asset.rack_id,
-                assetcp.rack_position,
-                assetcp.model.height,
-                asset_id=assetcp.id,
-            )
-        except LocationException:
-            assetcp.asset_conflict_location = asset
-            AssetCP.objects.filter(id=assetcp.id).update(asset_conflict_location=asset)
+    if asset.model.is_rackmount():
+        for assetcp in AssetCP.objects.filter(
```
Comment (juliasaveliff) [inline rackcity/utils/change_planner_utils.py:83]: would you mind renaming `assetcp` to `asset_cp`? it's more consistent with the rest of the code

File: rackcity/utils/errors_utils.py [hunk 3]
```diff
@@ -161,6 +161,11 @@ def parse_save_validation_error(error, object_name):
     failure_detail = ""
     if isinstance(error, ValidationError):
         for err in error:
+            print(err)
```
Comment (juliasaveliff) [inline rackcity/utils/errors_utils.py:164]: print statement :P

File: rackcity/utils/change_planner_utils.py (hunk 2 above)
Comment (benauriemma) [reply rackcity/utils/change_planner_utils.py:83]: I didn't write this but will do

File: rackcity/utils/errors_utils.py (hunk 3 above)
Comment (benauriemma) [reply rackcity/utils/errors_utils.py:164]: Damnnnn it

File: rackcity/models/asset.py (hunk 1 above)
Comment (benauriemma) [reply rackcity/models/asset.py:220]: `full_clean()` is essentially the validation method on a model -- it cleans the data for the database and will throw a ValidationError if any validation step fails.  We thought that it was called automatically on both serializer.save() and model.save(), but turns out it's only called on the serializer.  Read more about what it does here: https://docs.djangoproject.com/en/3.0/ref/models/instances/#validating-objects

Comment (juliasaveliff) [reply rackcity/models/asset.py:220]: HOW FUN!

File: rackcity/utils/change_planner_utils.py (hunk 2 above)
Comment (juliasaveliff) [reply rackcity/utils/change_planner_utils.py]: I thought so, just giving u extra work 😘

**Pull request merged**